onnx/onnx.proto

```diff
@@ -443,6 +448,21 @@ message TensorProto {
   repeated uint64 uint64_data = 11 [packed = true];
 }
 
+// A serialized sparse-tensor value
+message SparseTensorProto {
+  // The sequence of non-default values are encoded as a tensor of shape [NNZ].
+  // The default-value is zero for numeric tensors, and empty-string for string tensors.
+  optional TensorProto values = 1;
+
+  // The indices of the non-default values are encoded as a tensor of shape [NNZ].
+  // Note that the indices are linearized-indices. Each linearized index can be
+  // converted into a tuple of indices using the shape provided below.
+  optional TensorProto indices = 2;
```

Comment:
this should be just a repeated int64 indices?

Reply:
No. For a 2-D sparse tensor,
```
[ [0, 0],
  [1, 0] ],
```
the first element in `indices` would be `[1, 0]`.

Reply:
@linkerzhang : it is a int64 TensorProto, if that is your point. Using a TensorProto instead of "repeated int64" has a couple of advantages: one, as Wei-sheng mentions, it allows us to capture the "shape" of indices; second, it allows us to use the different formats supported by TensorProto for free.

Reply:
Fair enough.

Reply:
On second thoughts, I think it seems simpler to go with @linkerzhang 's suggestion of repeated int64 indices. (Of course, it will contain NNZ * rank entries, in the required order, which the spec should clarify.) It is unclear we will need the extra flexibility provided by the different formats in a TensorProto for a sparse tensor. So, I am inclined to change it as suggested.

Reply:
After more discussions, we decided to leave it as a TensorProto, given that protobuf repeated fields seem to use int rather than int64 for indexing. TensorProto's flexible options for representing data may end up being useful.